scripts/cbmc_json_parser.py

```diff
@@ -0,0 +1,409 @@
+
+            # Construct only summarized result and display output
+            output_message = construct_terse_property_message(properties)
+
+        # Print using an Interface function
+        print_to_terminal(output_message)
+
+        # Delete temp Json file after displaying result
+        clear_json_file(json_file)
+
+        # Check that temp files have been cleared
+        assert exists(json_file) == False, "Temp Files not cleared"
+    else:
+        # DynTrait tests generate a non json output due to "Invariant check failed" error
+        # For these cases, we just produce the cbmc output unparsed
+        # TODO: Parse these non json outputs from CBMC
+        non_json_cbmc_output_handler(log_file)
+    return
+
+def save_json_file(log_file):
+    """
+    Write the json file to disk for easy parsing
+    """
+
+    # TODO : Change to use the source file name as the temp file name too
+    # Create a temp json file for every source file as testing can be parallelized
+    tfile = tempfile.NamedTemporaryFile(prefix='cbmcparser_', suffix='.json').name
+    try:
+        with open(tfile, "w", encoding='utf-8') as outfile:
+            json.dump(log_file, outfile)
+    except FileNotFoundError as e:
+        print("Error in writing JSON file", e)
+    return tfile
+
+# Check if the blob is in json format for parsing
+def is_json(myjson):
+    try:
+        json_file = json.loads(myjson)
+    except ValueError:
+        return False, None
+    return True, json_file
+
+def print_json_file(input_json_file):
+    """
+    Function for printing temporary json objects for debugging purposes
+    """
+    with open(input_json_file) as infile:
+        data = json.load(infile)
+        print(data)
+    return
+
+def restructure_json_file(log_file):
+    """
+    CBMC's response blob is difficult to parse by itself, so it needs to be re-fit into
+    another temporary dictionary structure and
+
+    Parameters -
+        log_file - log_file is the initial cbmc blob object which even in JSON format,
+                    is difficult to parse and in some cases, is not in JSON format.
+
+    Returns -
+        temp_dict - The restructured dictionary which is easy to parse
+
+    """
+
+    # The restructured dictionary which is easy to parse
+    temp_dict = {}
+    try:
+        with open(log_file) as infile:
+            data = json.load(infile)
+            temp_dict["response"] = data
+    except FileNotFoundError:
+        print("CBMC File not found")
+        return {}
+    except JSONDecodeError:
+        # print out specific information about the extra data (like the lines before it)
+        print("JSON File not being parsed")
+        return {}
+
+    # TODO: Check if passing dictionaries around is optimal or if there are alternate ways to
+    # restructure and parse cbmc blob.
+    return temp_dict
+
+def extract_solver_information(json_object):
+    """
+    Get Information about the Solver from the Json file written
```

Comment:
This function does something important! It should get better docs. :)

Seems to be sorting these into two buckets based on whether `result` is a key.